{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### purpose upload fastq files to SRA\n",
    "\n",
    "\n",
    "### outline\n",
    "\n",
    "create files to upload to SRA, dryad, or for metadata (SRA, biosample). The script was run (without transferring to SRA) to create the biosample metadata to register the biosamples, the script was then rerun to add in biosample accession info before uploading to SRA.\n",
    "\n",
    "[get fastqfiles](#fastq)\n",
    "- for both Doug-fir and jack pine, get file paths to the fastq files I want to upload to SRA\n",
    "- use pipeline config files to get applicable file names, upload to SRA\n",
    "\n",
    "[check md5s](#check)\n",
    "- check md5 files for each of the fastq files, upload these to SRA\n",
    "\n",
    "[bgzip snp tables](#copy)\n",
    "- bgzip the .txt files representing the post-pipeline SNP files used in previous notebooks\n",
    "- upload these to SRA\n",
    "\n",
    "[prepare SRA metadata doc](#sradoc)\n",
    "- use this file to manually upload SRA metadata\n",
    "- add md5 hashes as columns for `filename` and `filename2`\n",
    "\n",
    "[prepare biosample metadata doc](#biosample)\n",
    "- using the SRA metadata doc, create the biosample metadata file to upload to NCBI\n",
    "\n",
    "[transfer files to SRA](#transfer)\n",
    "- transfer fastq and fastq.md5 files to SRA via ftp"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pythonimports import *\n",
    "import ftplib\n",
    "\n",
    "fqdir = '/data/fastq/mengmeng/SeqCap_Test_for_CoAdapTree'\n",
    "tmpdir = makedir('/lu213/brandon.lind/data/testdir/dryad_testdata')\n",
    "sra_main = '/data/projects/pool_seq/sra_docs'\n",
    "\n",
    "lview,dview = get_client()\n",
    "\n",
    "latest_commit()\n",
    "sinfo(html=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# a list for all files to transfer\n",
    "sra_transfer_files = []\n",
    "dryad = []"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id='fastq'></a>\n",
    "# get the fastq files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### pool and individual files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>library_name</th>\n",
       "      <th>pool_name</th>\n",
       "      <th>ploidy</th>\n",
       "      <th>file_name_r1</th>\n",
       "      <th>file_name_r2</th>\n",
       "      <th>adaptor_1</th>\n",
       "      <th>adaptor_2</th>\n",
       "      <th>ref</th>\n",
       "      <th>rgid</th>\n",
       "      <th>rglb</th>\n",
       "      <th>rgpl</th>\n",
       "      <th>rgpu</th>\n",
       "      <th>rgsm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DF_52_20</td>\n",
       "      <td>DF_cap3_kit1</td>\n",
       "      <td>DF_i52</td>\n",
       "      <td>2</td>\n",
       "      <td>HI.4779.008.D705---D505.DF_52_20_cap3_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D705---D505.DF_52_20_cap3_kit1_R2....</td>\n",
       "      <td>AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC</td>\n",
       "      <td>AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT</td>\n",
       "      <td>/project/def-saitken/lindb/refgenomes/DF_db/DF...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF.cap3.kit1</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_20_cap3_kit1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DF_52_21</td>\n",
       "      <td>DF_cap3_kit1</td>\n",
       "      <td>DF_i52</td>\n",
       "      <td>2</td>\n",
       "      <td>HI.4779.008.D706---D505.DF_52_21_cap3_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D706---D505.DF_52_21_cap3_kit1_R2....</td>\n",
       "      <td>AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC</td>\n",
       "      <td>AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT</td>\n",
       "      <td>/project/def-saitken/lindb/refgenomes/DF_db/DF...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF.cap3.kit1</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_21_cap3_kit1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DF_52_22</td>\n",
       "      <td>DF_cap3_kit1</td>\n",
       "      <td>DF_i52</td>\n",
       "      <td>2</td>\n",
       "      <td>HI.4779.008.D707---D505.DF_52_22_cap3_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D707---D505.DF_52_22_cap3_kit1_R2....</td>\n",
       "      <td>AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC</td>\n",
       "      <td>AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT</td>\n",
       "      <td>/project/def-saitken/lindb/refgenomes/DF_db/DF...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF.cap3.kit1</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_22_cap3_kit1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DF_52_23</td>\n",
       "      <td>DF_cap3_kit1</td>\n",
       "      <td>DF_i52</td>\n",
       "      <td>2</td>\n",
       "      <td>HI.4779.008.D708---D505.DF_52_23_cap3_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D708---D505.DF_52_23_cap3_kit1_R2....</td>\n",
       "      <td>AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC</td>\n",
       "      <td>AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT</td>\n",
       "      <td>/project/def-saitken/lindb/refgenomes/DF_db/DF...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF.cap3.kit1</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_23_cap3_kit1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DF_52_24</td>\n",
       "      <td>DF_cap3_kit1</td>\n",
       "      <td>DF_i52</td>\n",
       "      <td>2</td>\n",
       "      <td>HI.4779.008.D709---D505.DF_52_24_cap3_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D709---D505.DF_52_24_cap3_kit1_R2....</td>\n",
       "      <td>AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC</td>\n",
       "      <td>AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT</td>\n",
       "      <td>/project/def-saitken/lindb/refgenomes/DF_db/DF...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF.cap3.kit1</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_24_cap3_kit1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  sample_name  library_name pool_name  ploidy  \\\n",
       "0    DF_52_20  DF_cap3_kit1    DF_i52       2   \n",
       "1    DF_52_21  DF_cap3_kit1    DF_i52       2   \n",
       "2    DF_52_22  DF_cap3_kit1    DF_i52       2   \n",
       "3    DF_52_23  DF_cap3_kit1    DF_i52       2   \n",
       "4    DF_52_24  DF_cap3_kit1    DF_i52       2   \n",
       "\n",
       "                                        file_name_r1  \\\n",
       "0  HI.4779.008.D705---D505.DF_52_20_cap3_kit1_R1....   \n",
       "1  HI.4779.008.D706---D505.DF_52_21_cap3_kit1_R1....   \n",
       "2  HI.4779.008.D707---D505.DF_52_22_cap3_kit1_R1....   \n",
       "3  HI.4779.008.D708---D505.DF_52_23_cap3_kit1_R1....   \n",
       "4  HI.4779.008.D709---D505.DF_52_24_cap3_kit1_R1....   \n",
       "\n",
       "                                        file_name_r2  \\\n",
       "0  HI.4779.008.D705---D505.DF_52_20_cap3_kit1_R2....   \n",
       "1  HI.4779.008.D706---D505.DF_52_21_cap3_kit1_R2....   \n",
       "2  HI.4779.008.D707---D505.DF_52_22_cap3_kit1_R2....   \n",
       "3  HI.4779.008.D708---D505.DF_52_23_cap3_kit1_R2....   \n",
       "4  HI.4779.008.D709---D505.DF_52_24_cap3_kit1_R2....   \n",
       "\n",
       "                            adaptor_1                          adaptor_2  \\\n",
       "0  AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC  AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT   \n",
       "1  AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC  AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT   \n",
       "2  AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC  AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT   \n",
       "3  AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC  AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT   \n",
       "4  AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC  AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT   \n",
       "\n",
       "                                                 ref  rgid          rglb  \\\n",
       "0  /project/def-saitken/lindb/refgenomes/DF_db/DF...   NaN  DF.cap3.kit1   \n",
       "1  /project/def-saitken/lindb/refgenomes/DF_db/DF...   NaN  DF.cap3.kit1   \n",
       "2  /project/def-saitken/lindb/refgenomes/DF_db/DF...   NaN  DF.cap3.kit1   \n",
       "3  /project/def-saitken/lindb/refgenomes/DF_db/DF...   NaN  DF.cap3.kit1   \n",
       "4  /project/def-saitken/lindb/refgenomes/DF_db/DF...   NaN  DF.cap3.kit1   \n",
       "\n",
       "       rgpl  rgpu                rgsm  \n",
       "0  ILLUMINA   NaN  DF_52_20_cap3_kit1  \n",
       "1  ILLUMINA   NaN  DF_52_21_cap3_kit1  \n",
       "2  ILLUMINA   NaN  DF_52_22_cap3_kit1  \n",
       "3  ILLUMINA   NaN  DF_52_23_cap3_kit1  \n",
       "4  ILLUMINA   NaN  DF_52_24_cap3_kit1  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# infile/config to gatk pipeline for pool-seq and individual-seq\n",
    "datafile = '/data/projects/pool_seq/non-pangenome/gatk_diploid_testdata/JP_i101-gatk/datatable.txt'\n",
    "datatable = pd.read_table(datafile)\n",
    "datatable.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(84, 84)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get file paths to each of the files\n",
    "# 40 individual for each species (R1 + R2 * 20 samps * 2 species)\n",
    "# 2 poolseq files for each species (R1 + R2 * 1 samps * 2 species)\n",
    "# 40 + 2 + 40 + 2 = 84\n",
    "fastqs = []\n",
    "fqs = datatable['file_name_r1'].tolist() + datatable['file_name_r2'].tolist()\n",
    "for fq in fqs:\n",
    "    fastq = op.join(fqdir, fq)\n",
    "    assert op.exists(fastq)\n",
    "    fastqs.append(fastq)\n",
    "assert len(fastqs) == luni(fastqs)\n",
    "len(fastqs), luni(fastqs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "sra_transfer_files.extend(fastqs)\n",
    "sra_transfer_files.extend([f + '.md5' for f in fastqs])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "dryad.append(datafile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### mega files "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/data/fastq/mengmeng/Mega_RefSeq/HI.4992.002.D709---D502.JP_RFmg7_cap44_kit6_R1.fastq.gz',\n",
       " '/data/fastq/mengmeng/Mega_RefSeq/HI.4992.002.D709---D502.JP_RFmg7_cap44_kit6_R2.fastq.gz']"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get mega-seq files\n",
    "megadir = '/data/fastq/mengmeng/Mega_RefSeq'\n",
    "megafile = '/data/projects/pool_seq/non-pangenome/varscan_mega/JP_RFmg7/datatable.txt'\n",
    "megadt = pd.read_table(megafile)\n",
    "megadt = megadt[megadt['sample_name'] == 'JP_RFmg7']\n",
    "mega_fastqs = [op.join(megadir, f) for f in megadt['file_name_r1'].tolist() + megadt['file_name_r2'].tolist()]\n",
    "mega_fastqs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "sra_transfer_files.extend(mega_fastqs)\n",
    "sra_transfer_files.extend([f + '.md5' for f in mega_fastqs])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "dryad.append(megafile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id='check'></a>\n",
    "# check md5s"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def check_md5(f):\n",
    "    import shutil, subprocess, os\n",
    "    from os import path as op\n",
    "    \n",
    "    os.chdir(op.dirname(f))\n",
    "    \n",
    "    md5 = f + '.md5'\n",
    "    assert op.exists(md5)\n",
    "    \n",
    "    stdout = subprocess.check_output(\n",
    "        [shutil.which('md5sum'),\n",
    "         '-c',\n",
    "         md5\n",
    "        ]\n",
    "    )\n",
    "    \n",
    "    return stdout"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1m\n",
      "Watching 86 jobs ...\u001b[0m\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 86/86 [00:25<00:00,  3.38it/s]\n"
     ]
    }
   ],
   "source": [
    "# double check md5 of existing .md5 files in parallel\n",
    "jobs = []\n",
    "for fastq in fastqs + mega_fastqs:\n",
    "    jobs.append(\n",
    "        lview.apply_async(\n",
    "            check_md5, fastq\n",
    "        )\n",
    "    )\n",
    "    \n",
    "watch_async(jobs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'OK'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# make sure all of the ind-seq/pool-seq md5 files check out\n",
    "for j in jobs:\n",
    "    status = j.r\\\n",
    "            .decode('utf-8')\\\n",
    "            .split('\\n')[0]\\\n",
    "            .rstrip('\\n')\\\n",
    "            .split()[-1]\n",
    "    assert status == 'OK'\n",
    "status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# map file to md5 hash\n",
    "md5s = {}\n",
    "for f in fastqs + mega_fastqs:\n",
    "    md5 = f + '.md5'\n",
    "    assert op.exists(md5)\n",
    "    md5hash = read(md5)[0].split()[0]\n",
    "    md5s[op.basename(f)] = md5hash\n",
    "\n",
    "assert len(md5s) == len(fastqs) + len(mega_fastqs)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id='copy'></a>\n",
    "# bgzip snp tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "def copy_and_bgzip(src, dst):\n",
    "    import shutil, subprocess\n",
    "    \n",
    "    shutil.copy(src, dst)\n",
    "    \n",
    "    stdout = subprocess.check_output(\n",
    "        [shutil.which('bgzip'),\n",
    "         '-f',\n",
    "         dst\n",
    "        ]\n",
    "    )\n",
    "    \n",
    "    return stdout"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "snptables = {\n",
    "    'mega' : {\n",
    "        'JP': op.join('/data/projects/pool_seq/non-pangenome/varscan_mega/JP_RFmg7/snpsANDindels',\n",
    "                      'JP_RFmg7-varscan_all_bedfiles_SNP_paralog_snps_translated.txt'),\n",
    "        'DF': op.join('/data/projects/pool_seq/non-pangenome/varscan_mega/DF_megaSNPs_from_download/DF_mega',\n",
    "                      'snpsANDindels/02_baseline_filtered/DF_mega-varscan_all_bedfiles_SNP_paralog_snps.txt')},\n",
    "    'gatk' : {\n",
    "        'JP': op.join('/data/projects/pool_seq/non-pangenome/gatk_diploid_testdata/JP_i101-gatk/filtered_snps',\n",
    "                      'JP_i101_filtered_concatenated_snps_max-missing_table_biallelic-only_translated.txt'),\n",
    "        'DF': op.join('/data/projects/pool_seq/non-pangenome/gatk_diploid_testdata/DF_i52-gatk/filtered_snps',\n",
    "                      'DF_i52_filtered_concatenated_snps_max-missing_table_biallelic-only_p000_translated.txt')},\n",
    "    'varscan' : {\n",
    "        'JP': op.join('/data/projects/pool_seq/non-pangenome/varscan_pooled/JP_pooled/snpsANDindels',\n",
    "                      'JP_pooled-varscan_all_bedfiles_SNP_translated.txt'),\n",
    "        'DF': op.join('/data/projects/pool_seq/non-pangenome/varscan_pooled/DF_p52/snpsANDindels',\n",
    "                      'DF_p52-varscan_all_bedfiles_SNP.txt')\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1m\n",
      "Watching 6 jobs ...\u001b[0m\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 6/6 [07:02<00:00, 70.37s/it] \n"
     ]
    }
   ],
   "source": [
    "# copy snp tables to new folder, bgzip in parallel\n",
    "jobs = []\n",
    "for source,sppdict in snptables.items():\n",
    "    for spp,snptable in sppdict.items():\n",
    "        dst = op.join(tmpdir, op.basename(snptable))\n",
    "        if op.exists(dst + '.gz'):\n",
    "            continue\n",
    "        jobs.append(\n",
    "            lview.apply_async(\n",
    "                copy_and_bgzip, *(snptable, dst)\n",
    "            )\n",
    "        )\n",
    "watch_async(jobs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get the bgzipped files\n",
    "snpgzs = fs(tmpdir, endswith='.gz')\n",
    "len(snpgzs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "dryad.extend(snpgzs)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id='sradoc'></a>\n",
    "# prepare SRA doc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biosample_accession</th>\n",
       "      <th>library_ID</th>\n",
       "      <th>title</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>platform</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>design_description</th>\n",
       "      <th>filetype</th>\n",
       "      <th>filename</th>\n",
       "      <th>filename2</th>\n",
       "      <th>filename3</th>\n",
       "      <th>filename4</th>\n",
       "      <th>assembly</th>\n",
       "      <th>fasta_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [biosample_accession, library_ID, title, library_strategy, library_source, library_selection, library_layout, platform, instrument_model, design_description, filetype, filename, filename2, filename3, filename4, assembly, fasta_file]\n",
       "Index: []"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load SRA metadata template file\n",
    "sra_doc = pd.read_table(op.join(sra_main, 'SRA_metadata_acc.txt')).loc[range(0)]\n",
    "sra_doc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_to_table(df):\n",
    "    sra_tmp = pd.read_table(op.join(sra_main, 'SRA_metadata_acc.txt')).loc[range(0)]\n",
    "    for row in df.index:\n",
    "        samp,group,r1,r2 = df.loc[row, ['sample_name', 'pool_name', 'file_name_r1', 'file_name_r2']]\n",
    "        \n",
    "        sra_tmp.loc[row, ['filename', 'filename2']] = r1,r2\n",
    "\n",
    "        if 'mg' in samp:\n",
    "            tissue_type = 'haploid megagametophyte'\n",
    "        else:\n",
    "            tissue_type = 'diploid needle'\n",
    "\n",
    "        if 'JP' in samp:\n",
    "            spp = 'jack pine (Pinus banksiana)'\n",
    "        else:\n",
    "            spp = 'Douglas-fir (Pseudotsuga menziesii)'\n",
    "\n",
    "        if '_p' in samp:\n",
    "            ploidy = 'pooled diploid'\n",
    "        else:\n",
    "            ploidy = 'individual diploid'\n",
    "        if tissue_type == 'haploid megagametophyte':\n",
    "            ploidy = 'haploid'\n",
    "\n",
    "\n",
    "        sra_tmp.loc[row, 'library_ID'] = samp\n",
    "        sra_tmp.loc[row, 'title'] = f'{ploidy} exome-capture of {spp} : {tissue_type} tissue'\n",
    "    \n",
    "    sra_tmp['design_description'] = 'exome sequences were isolatated using custom targeted sequence capture probes'\n",
    "    sra_tmp['library_source'] = 'GENOMIC'\n",
    "    sra_tmp['library_selection'] = 'PCR'\n",
    "    sra_tmp['library_layout'] = 'Paired'\n",
    "    sra_tmp['platform'] = 'Illumina'\n",
    "    sra_tmp['instrument_model'] = 'Illumina HiSeq 4000'\n",
    "    sra_tmp['library_strategy'] = 'OTHER'\n",
    "    sra_tmp['filetype'] = 'fastq'\n",
    "    \n",
    "    return sra_tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'DF_52_20': 'SAMN20086271', 'DF_52_21': 'SAMN20086272', 'DF_52_22': 'SAMN20086273', 'DF_52_23': 'SAMN20086274', 'DF_52_24': 'SAMN20086275', 'DF_52_25': 'SAMN20086276', 'DF_52_26': 'SAMN20086277', 'DF_52_27': 'SAMN20086278', 'DF_52_28': 'SAMN20086279', 'DF_52_29': 'SAMN20086280', 'DF_52_30': 'SAMN20086281', 'DF_52_31': 'SAMN20086282', 'DF_52_32': 'SAMN20086283', 'DF_52_33': 'SAMN20086284', 'DF_52_34': 'SAMN20086285', 'DF_52_35': 'SAMN20086286', 'DF_52_36': 'SAMN20086287', 'DF_52_37': 'SAMN20086288', 'DF_52_38': 'SAMN20086289', 'DF_52_39': 'SAMN20086290', 'DF_p52': 'SAMN20086291', 'JP_101_1': 'SAMN20086292', 'JP_101_2': 'SAMN20086293', 'JP_101_3': 'SAMN20086294', 'JP_101_4': 'SAMN20086295', 'JP_101_5': 'SAMN20086296', 'JP_101_6': 'SAMN20086297', 'JP_101_7': 'SAMN20086298', 'JP_101_8': 'SAMN20086299', 'JP_101_9': 'SAMN20086300', 'JP_101_10': 'SAMN20086301', 'JP_101_11': 'SAMN20086302', 'JP_101_12': 'SAMN20086303', 'JP_101_13': 'SAMN20086304', 'JP_101_14': 'SAMN20086305', 'JP_101_15': 'SAMN20086306', 'JP_101_16': 'SAMN20086307', 'JP_101_17': 'SAMN20086308', 'JP_101_18': 'SAMN20086309', 'JP_101_19': 'SAMN20086310', 'JP_101_20': 'SAMN20086311', 'JP_p101': 'SAMN20086312', 'JP_RFmg7': 'SAMN20086313'}\n"
     ]
    }
   ],
   "source": [
    "sra_dir = makedir(op.join(sra_main, 'testdata_submission'))\n",
    "# tsv created by registering samples with biosample metadata below\n",
    "# use to map our pop ID to biosample_accession\n",
    "biosamplefile = op.join(sra_dir, 'biosample_attributes.tsv')\n",
    "biodf = pd.read_table(biosamplefile)\n",
    "biodict = dict(zip(biodf['sample_name'], biodf['accession']))\n",
    "print(biodict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biosample_accession</th>\n",
       "      <th>library_ID</th>\n",
       "      <th>title</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>platform</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>design_description</th>\n",
       "      <th>filetype</th>\n",
       "      <th>filename</th>\n",
       "      <th>filename2</th>\n",
       "      <th>filename3</th>\n",
       "      <th>filename4</th>\n",
       "      <th>assembly</th>\n",
       "      <th>fasta_file</th>\n",
       "      <th>md5sum_filename</th>\n",
       "      <th>md5sum_filename2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMN20086271</td>\n",
       "      <td>DF_52_20</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D705---D505.DF_52_20_cap3_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D705---D505.DF_52_20_cap3_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>e5d15b40d1854dbe22f80fa424a950a3</td>\n",
       "      <td>ee60808baf5be0efead6d98ff4df8b57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMN20086272</td>\n",
       "      <td>DF_52_21</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D706---D505.DF_52_21_cap3_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D706---D505.DF_52_21_cap3_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>dde217cf48666870be64700eb9c0c609</td>\n",
       "      <td>5c5fcfc1aabfd3bd585b82accfacc7cb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SAMN20086273</td>\n",
       "      <td>DF_52_22</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D707---D505.DF_52_22_cap3_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D707---D505.DF_52_22_cap3_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ff3c134c6e9b973f52435f9f6bf40b46</td>\n",
       "      <td>cb006b5bcf0bfbf2fc2db7d47884d00b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SAMN20086274</td>\n",
       "      <td>DF_52_23</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D708---D505.DF_52_23_cap3_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D708---D505.DF_52_23_cap3_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5405160b462c2bb8cc1402b001752030</td>\n",
       "      <td>0530c698b878fa5072be186ab376c299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAMN20086275</td>\n",
       "      <td>DF_52_24</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D709---D505.DF_52_24_cap3_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D709---D505.DF_52_24_cap3_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>a26801d87bed5cadd1e144655f557fa3</td>\n",
       "      <td>ccf722a07c85e4f7dd23e78ad887b5a1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SAMN20086276</td>\n",
       "      <td>DF_52_25</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D710---D505.DF_52_25_cap3_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D710---D505.DF_52_25_cap3_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fbfe8a79f4cd7d43e3241daf0631b436</td>\n",
       "      <td>5cef6d35bf4f78fc7f2cf93dce81b176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SAMN20086277</td>\n",
       "      <td>DF_52_26</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D711---D505.DF_52_26_cap4_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D711---D505.DF_52_26_cap4_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>56dc37f3687405ca9c4384fe6d52d5d7</td>\n",
       "      <td>07d132123750d7f9a10b4a48efc8da2e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SAMN20086278</td>\n",
       "      <td>DF_52_27</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D712---D505.DF_52_27_cap4_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D712---D505.DF_52_27_cap4_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4439283a754d92079df65a8e1009286c</td>\n",
       "      <td>d7e67ea063d959210e95f14d93b8fc06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SAMN20086279</td>\n",
       "      <td>DF_52_28</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D701---D506.DF_52_28_cap4_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D701---D506.DF_52_28_cap4_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0ca31d096ac7b5b276fc34c45fa04805</td>\n",
       "      <td>1cab668b96a996ffd7c09f2d2dfe02d4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SAMN20086280</td>\n",
       "      <td>DF_52_29</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D702---D506.DF_52_29_cap4_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D702---D506.DF_52_29_cap4_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3f2c9090a11bcc406ae47e95c79d10d6</td>\n",
       "      <td>15ed66bb31debe7ad1487bf225b8e664</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SAMN20086281</td>\n",
       "      <td>DF_52_30</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D703---D506.DF_52_30_cap4_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D703---D506.DF_52_30_cap4_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>df0c42861313609155ebe1c1e764fefc</td>\n",
       "      <td>19340865fc53e595e63d7b02b723c1c1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SAMN20086282</td>\n",
       "      <td>DF_52_31</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D704---D506.DF_52_31_cap4_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D704---D506.DF_52_31_cap4_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cf46930abd6d2ad1f726baa37b42e87a</td>\n",
       "      <td>220563fd76c44abc3df3301f7a38b916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SAMN20086283</td>\n",
       "      <td>DF_52_32</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D705---D506.DF_52_32_cap4_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D705---D506.DF_52_32_cap4_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7aa3864d5e0f272bdb7c423654f6a456</td>\n",
       "      <td>aa26977b5946e17141ba81e84c973c94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMN20086284</td>\n",
       "      <td>DF_52_33</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D706---D506.DF_52_33_cap5_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D706---D506.DF_52_33_cap5_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3d04630962d63893dbe787553b9c3f51</td>\n",
       "      <td>039409788a4ec68c71fe60c575b049ac</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SAMN20086285</td>\n",
       "      <td>DF_52_34</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D707---D506.DF_52_34_cap5_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D707---D506.DF_52_34_cap5_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>70a5de1f4b4073dff74fff7258f0ddbf</td>\n",
       "      <td>c488d1236bf60fb09f5f58241b0d1811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SAMN20086286</td>\n",
       "      <td>DF_52_35</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D708---D506.DF_52_35_cap5_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D708---D506.DF_52_35_cap5_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>a17e83687aee0e018243eee2cc1a6ba4</td>\n",
       "      <td>7214ae872d9ebf3877a5034939e401f7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SAMN20086287</td>\n",
       "      <td>DF_52_36</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D709---D506.DF_52_36_cap5_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D709---D506.DF_52_36_cap5_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>b2ab6085a3e5975c07e9ca8667f67879</td>\n",
       "      <td>b7973b6ab309998bd78f73eef91c2ad6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SAMN20086288</td>\n",
       "      <td>DF_52_37</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D710---D506.DF_52_37_cap5_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D710---D506.DF_52_37_cap5_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0812d4fa8b503c47999b55ae670a8f27</td>\n",
       "      <td>600ea0e3e27810c661f4faa4bc4ce42e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>SAMN20086289</td>\n",
       "      <td>DF_52_38</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D711---D506.DF_52_38_cap5_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D711---D506.DF_52_38_cap5_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>56504de256a915bb62b2a0cc43269ec4</td>\n",
       "      <td>5d42b8f35203e4707d24372c55c9456b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SAMN20086290</td>\n",
       "      <td>DF_52_39</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D712---D506.DF_52_39_cap5_kit1_R1....</td>\n",
       "      <td>HI.4779.008.D712---D506.DF_52_39_cap5_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>762891a748a1943baeaca449905652d2</td>\n",
       "      <td>d1880109e1dfaace8f3f8049c32d771d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>SAMN20086291</td>\n",
       "      <td>DF_p52</td>\n",
       "      <td>pooled diploid exome-capture of Douglas-fir (P...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.008.D701---D507.DF_p52_cap1_kit2_R1.fa...</td>\n",
       "      <td>HI.4779.008.D701---D507.DF_p52_cap1_kit2_R2.fa...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>93a7e9b5697bf934e77831c1a0485002</td>\n",
       "      <td>f753734200ce808bc1a7e5df192813a1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SAMN20086292</td>\n",
       "      <td>JP_101_1</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D703---D507.JP_101_1_cap3_kit1_R1....</td>\n",
       "      <td>HI.4779.007.D703---D507.JP_101_1_cap3_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ad082957f1fc4b95c69d2b7bdf1ee31c</td>\n",
       "      <td>bf593c677390d9007294846c84cab753</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>SAMN20086293</td>\n",
       "      <td>JP_101_2</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D704---D507.JP_101_2_cap3_kit1_R1....</td>\n",
       "      <td>HI.4779.007.D704---D507.JP_101_2_cap3_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54fa6e041de38b722d55d90e231cec18</td>\n",
       "      <td>2783adfa0851a4ef68a12f0f008e1e95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SAMN20086294</td>\n",
       "      <td>JP_101_3</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D705---D507.JP_101_3_cap3_kit1_R1....</td>\n",
       "      <td>HI.4779.007.D705---D507.JP_101_3_cap3_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>db555bc3e3045033328a7a3090127434</td>\n",
       "      <td>68667446f1306001012d80814730f16c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>SAMN20086295</td>\n",
       "      <td>JP_101_4</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D706---D507.JP_101_4_cap3_kit1_R1....</td>\n",
       "      <td>HI.4779.007.D706---D507.JP_101_4_cap3_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6e3445a2cc22d242b3303bc85388aeeb</td>\n",
       "      <td>8eac74240e5465ac241e0b952f5f9001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>SAMN20086296</td>\n",
       "      <td>JP_101_5</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D707---D507.JP_101_5_cap3_kit1_R1....</td>\n",
       "      <td>HI.4779.007.D707---D507.JP_101_5_cap3_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>df6e4c0ed86662e485c53b250dfb1e11</td>\n",
       "      <td>56ae69379a44e7363e5cf15577436868</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>SAMN20086297</td>\n",
       "      <td>JP_101_6</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D708---D507.JP_101_6_cap3_kit1_R1....</td>\n",
       "      <td>HI.4779.007.D708---D507.JP_101_6_cap3_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2f40fdb00ec35e0382eab86f8196f16e</td>\n",
       "      <td>0bff2daf1fedb47dd33946146d725423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>SAMN20086298</td>\n",
       "      <td>JP_101_7</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D709---D507.JP_101_7_cap4_kit1_R1....</td>\n",
       "      <td>HI.4779.007.D709---D507.JP_101_7_cap4_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>67ab9a3edb538103b7a604b1bf6ae1b9</td>\n",
       "      <td>7b550ab972d4eb3323398b7397a30889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>SAMN20086299</td>\n",
       "      <td>JP_101_8</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D710---D507.JP_101_8_cap4_kit1_R1....</td>\n",
       "      <td>HI.4779.007.D710---D507.JP_101_8_cap4_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>80df2eeefa4e4366f0c95710c01aa54c</td>\n",
       "      <td>bd1fc32bf3d2f4d0d45863ad1b7eda97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>SAMN20086300</td>\n",
       "      <td>JP_101_9</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D711---D507.JP_101_9_cap4_kit1_R1....</td>\n",
       "      <td>HI.4779.007.D711---D507.JP_101_9_cap4_kit1_R2....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>130d7fd0dc1414ff2b21508ada630d4b</td>\n",
       "      <td>27ceada02aa0446d4dc6aed4c64fa4b6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>SAMN20086301</td>\n",
       "      <td>JP_101_10</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D712---D507.JP_101_10_cap4_kit1_R1...</td>\n",
       "      <td>HI.4779.007.D712---D507.JP_101_10_cap4_kit1_R2...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75cc8f8d6840c0a3c1fe115dbab8f6aa</td>\n",
       "      <td>f68e83b5443fe2ebf29dbab620627f7d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>SAMN20086302</td>\n",
       "      <td>JP_101_11</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D701---D508.JP_101_11_cap4_kit1_R1...</td>\n",
       "      <td>HI.4779.007.D701---D508.JP_101_11_cap4_kit1_R2...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>aabc3f5d1d747f2dbb3108c26b893791</td>\n",
       "      <td>933ae114dfc9a9761243bbbdd6378822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>SAMN20086303</td>\n",
       "      <td>JP_101_12</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D702---D508.JP_101_12_cap4_kit1_R1...</td>\n",
       "      <td>HI.4779.007.D702---D508.JP_101_12_cap4_kit1_R2...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>67caae3fea225c7a807097c161579d05</td>\n",
       "      <td>fd2b666f6ee543b5305725034d8c7257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>SAMN20086304</td>\n",
       "      <td>JP_101_13</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D703---D508.JP_101_13_cap4_kit1_R1...</td>\n",
       "      <td>HI.4779.007.D703---D508.JP_101_13_cap4_kit1_R2...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>f270e4762e964ee2647e9d6f23390e3b</td>\n",
       "      <td>ab37fd98d1578ae643a9c14f56a14ffb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>SAMN20086305</td>\n",
       "      <td>JP_101_14</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D707---D508.JP_101_14_cap5_kit1_R1...</td>\n",
       "      <td>HI.4779.007.D707---D508.JP_101_14_cap5_kit1_R2...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>d72ea8a81ba04e82de6c07dd16f89a69</td>\n",
       "      <td>1ae31f0f14de3e90ece90991ef49ebfb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>SAMN20086306</td>\n",
       "      <td>JP_101_15</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D704---D508.JP_101_15_cap5_kit1_R1...</td>\n",
       "      <td>HI.4779.007.D704---D508.JP_101_15_cap5_kit1_R2...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c6a7803776c8d9f23c7e61756a008ff2</td>\n",
       "      <td>a0400ecb78cf8374626726500a5525a6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>SAMN20086307</td>\n",
       "      <td>JP_101_16</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D705---D508.JP_101_16_cap5_kit1_R1...</td>\n",
       "      <td>HI.4779.007.D705---D508.JP_101_16_cap5_kit1_R2...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9d63dfca7cd51cf6139b933469d3c7d0</td>\n",
       "      <td>da8aa4ae001507eb3e23a7d34410b4b6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>SAMN20086308</td>\n",
       "      <td>JP_101_17</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D706---D508.JP_101_17_cap5_kit1_R1...</td>\n",
       "      <td>HI.4779.007.D706---D508.JP_101_17_cap5_kit1_R2...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>85cf94e964429e966847aeef718dc41e</td>\n",
       "      <td>4f44a1b88eed7c0eb0c5cfb30fba98c1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>SAMN20086309</td>\n",
       "      <td>JP_101_18</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D708---D508.JP_101_18_cap5_kit1_R1...</td>\n",
       "      <td>HI.4779.007.D708---D508.JP_101_18_cap5_kit1_R2...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9c3c040fcc745012119a7e1d4660e167</td>\n",
       "      <td>0c0e24d8bbb4793f976fb43e40d3ba86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>SAMN20086310</td>\n",
       "      <td>JP_101_19</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D709---D508.JP_101_19_cap5_kit1_R1...</td>\n",
       "      <td>HI.4779.007.D709---D508.JP_101_19_cap5_kit1_R2...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>309f0ad70d7dd49786dd3cf6b5375cc7</td>\n",
       "      <td>b1cd39cae55e2e3fa5f1380bcc717f07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>SAMN20086311</td>\n",
       "      <td>JP_101_20</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D710---D508.JP_101_20_cap5_kit1_R1...</td>\n",
       "      <td>HI.4779.007.D710---D508.JP_101_20_cap5_kit1_R2...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>360b8921683861ad16f9c6a9e68f3f89</td>\n",
       "      <td>5517ac0f913201b94950a8f19fea395d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>SAMN20086312</td>\n",
       "      <td>JP_p101</td>\n",
       "      <td>pooled diploid exome-capture of jack pine (Pin...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4779.007.D711---D508.JP_p101_cap1_kit1_R1.f...</td>\n",
       "      <td>HI.4779.007.D711---D508.JP_p101_cap1_kit1_R2.f...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c78ef4f756daef61c9b2ff1823715dd9</td>\n",
       "      <td>b6d7c918716125f052e67fda80575b78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>SAMN20086313</td>\n",
       "      <td>JP_RFmg7</td>\n",
       "      <td>haploid exome-capture of jack pine (Pinus bank...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Paired</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>exome sequences were isolatated using custom t...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HI.4992.002.D709---D502.JP_RFmg7_cap44_kit6_R1...</td>\n",
       "      <td>HI.4992.002.D709---D502.JP_RFmg7_cap44_kit6_R2...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cb94de1a82491caf1e62fe4792cb7cab</td>\n",
       "      <td>342e45d7c4eb19fb61256997209c2cdb</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   biosample_accession library_ID  \\\n",
       "0         SAMN20086271   DF_52_20   \n",
       "1         SAMN20086272   DF_52_21   \n",
       "2         SAMN20086273   DF_52_22   \n",
       "3         SAMN20086274   DF_52_23   \n",
       "4         SAMN20086275   DF_52_24   \n",
       "5         SAMN20086276   DF_52_25   \n",
       "6         SAMN20086277   DF_52_26   \n",
       "7         SAMN20086278   DF_52_27   \n",
       "8         SAMN20086279   DF_52_28   \n",
       "9         SAMN20086280   DF_52_29   \n",
       "10        SAMN20086281   DF_52_30   \n",
       "11        SAMN20086282   DF_52_31   \n",
       "12        SAMN20086283   DF_52_32   \n",
       "13        SAMN20086284   DF_52_33   \n",
       "14        SAMN20086285   DF_52_34   \n",
       "15        SAMN20086286   DF_52_35   \n",
       "16        SAMN20086287   DF_52_36   \n",
       "17        SAMN20086288   DF_52_37   \n",
       "18        SAMN20086289   DF_52_38   \n",
       "19        SAMN20086290   DF_52_39   \n",
       "20        SAMN20086291     DF_p52   \n",
       "21        SAMN20086292   JP_101_1   \n",
       "22        SAMN20086293   JP_101_2   \n",
       "23        SAMN20086294   JP_101_3   \n",
       "24        SAMN20086295   JP_101_4   \n",
       "25        SAMN20086296   JP_101_5   \n",
       "26        SAMN20086297   JP_101_6   \n",
       "27        SAMN20086298   JP_101_7   \n",
       "28        SAMN20086299   JP_101_8   \n",
       "29        SAMN20086300   JP_101_9   \n",
       "30        SAMN20086301  JP_101_10   \n",
       "31        SAMN20086302  JP_101_11   \n",
       "32        SAMN20086303  JP_101_12   \n",
       "33        SAMN20086304  JP_101_13   \n",
       "34        SAMN20086305  JP_101_14   \n",
       "35        SAMN20086306  JP_101_15   \n",
       "36        SAMN20086307  JP_101_16   \n",
       "37        SAMN20086308  JP_101_17   \n",
       "38        SAMN20086309  JP_101_18   \n",
       "39        SAMN20086310  JP_101_19   \n",
       "40        SAMN20086311  JP_101_20   \n",
       "41        SAMN20086312    JP_p101   \n",
       "42        SAMN20086313   JP_RFmg7   \n",
       "\n",
       "                                                title library_strategy  \\\n",
       "0   individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "1   individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "2   individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "3   individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "4   individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "5   individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "6   individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "7   individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "8   individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "9   individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "10  individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "11  individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "12  individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "13  individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "14  individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "15  individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "16  individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "17  individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "18  individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "19  individual diploid exome-capture of Douglas-fi...            OTHER   \n",
       "20  pooled diploid exome-capture of Douglas-fir (P...            OTHER   \n",
       "21  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "22  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "23  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "24  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "25  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "26  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "27  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "28  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "29  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "30  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "31  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "32  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "33  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "34  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "35  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "36  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "37  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "38  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "39  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "40  individual diploid exome-capture of jack pine ...            OTHER   \n",
       "41  pooled diploid exome-capture of jack pine (Pin...            OTHER   \n",
       "42  haploid exome-capture of jack pine (Pinus bank...            OTHER   \n",
       "\n",
       "   library_source library_selection library_layout  platform  \\\n",
       "0         GENOMIC               PCR         Paired  Illumina   \n",
       "1         GENOMIC               PCR         Paired  Illumina   \n",
       "2         GENOMIC               PCR         Paired  Illumina   \n",
       "3         GENOMIC               PCR         Paired  Illumina   \n",
       "4         GENOMIC               PCR         Paired  Illumina   \n",
       "5         GENOMIC               PCR         Paired  Illumina   \n",
       "6         GENOMIC               PCR         Paired  Illumina   \n",
       "7         GENOMIC               PCR         Paired  Illumina   \n",
       "8         GENOMIC               PCR         Paired  Illumina   \n",
       "9         GENOMIC               PCR         Paired  Illumina   \n",
       "10        GENOMIC               PCR         Paired  Illumina   \n",
       "11        GENOMIC               PCR         Paired  Illumina   \n",
       "12        GENOMIC               PCR         Paired  Illumina   \n",
       "13        GENOMIC               PCR         Paired  Illumina   \n",
       "14        GENOMIC               PCR         Paired  Illumina   \n",
       "15        GENOMIC               PCR         Paired  Illumina   \n",
       "16        GENOMIC               PCR         Paired  Illumina   \n",
       "17        GENOMIC               PCR         Paired  Illumina   \n",
       "18        GENOMIC               PCR         Paired  Illumina   \n",
       "19        GENOMIC               PCR         Paired  Illumina   \n",
       "20        GENOMIC               PCR         Paired  Illumina   \n",
       "21        GENOMIC               PCR         Paired  Illumina   \n",
       "22        GENOMIC               PCR         Paired  Illumina   \n",
       "23        GENOMIC               PCR         Paired  Illumina   \n",
       "24        GENOMIC               PCR         Paired  Illumina   \n",
       "25        GENOMIC               PCR         Paired  Illumina   \n",
       "26        GENOMIC               PCR         Paired  Illumina   \n",
       "27        GENOMIC               PCR         Paired  Illumina   \n",
       "28        GENOMIC               PCR         Paired  Illumina   \n",
       "29        GENOMIC               PCR         Paired  Illumina   \n",
       "30        GENOMIC               PCR         Paired  Illumina   \n",
       "31        GENOMIC               PCR         Paired  Illumina   \n",
       "32        GENOMIC               PCR         Paired  Illumina   \n",
       "33        GENOMIC               PCR         Paired  Illumina   \n",
       "34        GENOMIC               PCR         Paired  Illumina   \n",
       "35        GENOMIC               PCR         Paired  Illumina   \n",
       "36        GENOMIC               PCR         Paired  Illumina   \n",
       "37        GENOMIC               PCR         Paired  Illumina   \n",
       "38        GENOMIC               PCR         Paired  Illumina   \n",
       "39        GENOMIC               PCR         Paired  Illumina   \n",
       "40        GENOMIC               PCR         Paired  Illumina   \n",
       "41        GENOMIC               PCR         Paired  Illumina   \n",
       "42        GENOMIC               PCR         Paired  Illumina   \n",
       "\n",
       "       instrument_model                                 design_description  \\\n",
       "0   Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "1   Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "2   Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "3   Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "4   Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "5   Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "6   Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "7   Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "8   Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "9   Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "10  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "11  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "12  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "13  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "14  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "15  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "16  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "17  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "18  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "19  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "20  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "21  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "22  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "23  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "24  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "25  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "26  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "27  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "28  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "29  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "30  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "31  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "32  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "33  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "34  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "35  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "36  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "37  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "38  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "39  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "40  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "41  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "42  Illumina HiSeq 4000  exome sequences were isolatated using custom t...   \n",
       "\n",
       "   filetype                                           filename  \\\n",
       "0     fastq  HI.4779.008.D705---D505.DF_52_20_cap3_kit1_R1....   \n",
       "1     fastq  HI.4779.008.D706---D505.DF_52_21_cap3_kit1_R1....   \n",
       "2     fastq  HI.4779.008.D707---D505.DF_52_22_cap3_kit1_R1....   \n",
       "3     fastq  HI.4779.008.D708---D505.DF_52_23_cap3_kit1_R1....   \n",
       "4     fastq  HI.4779.008.D709---D505.DF_52_24_cap3_kit1_R1....   \n",
       "5     fastq  HI.4779.008.D710---D505.DF_52_25_cap3_kit1_R1....   \n",
       "6     fastq  HI.4779.008.D711---D505.DF_52_26_cap4_kit1_R1....   \n",
       "7     fastq  HI.4779.008.D712---D505.DF_52_27_cap4_kit1_R1....   \n",
       "8     fastq  HI.4779.008.D701---D506.DF_52_28_cap4_kit1_R1....   \n",
       "9     fastq  HI.4779.008.D702---D506.DF_52_29_cap4_kit1_R1....   \n",
       "10    fastq  HI.4779.008.D703---D506.DF_52_30_cap4_kit1_R1....   \n",
       "11    fastq  HI.4779.008.D704---D506.DF_52_31_cap4_kit1_R1....   \n",
       "12    fastq  HI.4779.008.D705---D506.DF_52_32_cap4_kit1_R1....   \n",
       "13    fastq  HI.4779.008.D706---D506.DF_52_33_cap5_kit1_R1....   \n",
       "14    fastq  HI.4779.008.D707---D506.DF_52_34_cap5_kit1_R1....   \n",
       "15    fastq  HI.4779.008.D708---D506.DF_52_35_cap5_kit1_R1....   \n",
       "16    fastq  HI.4779.008.D709---D506.DF_52_36_cap5_kit1_R1....   \n",
       "17    fastq  HI.4779.008.D710---D506.DF_52_37_cap5_kit1_R1....   \n",
       "18    fastq  HI.4779.008.D711---D506.DF_52_38_cap5_kit1_R1....   \n",
       "19    fastq  HI.4779.008.D712---D506.DF_52_39_cap5_kit1_R1....   \n",
       "20    fastq  HI.4779.008.D701---D507.DF_p52_cap1_kit2_R1.fa...   \n",
       "21    fastq  HI.4779.007.D703---D507.JP_101_1_cap3_kit1_R1....   \n",
       "22    fastq  HI.4779.007.D704---D507.JP_101_2_cap3_kit1_R1....   \n",
       "23    fastq  HI.4779.007.D705---D507.JP_101_3_cap3_kit1_R1....   \n",
       "24    fastq  HI.4779.007.D706---D507.JP_101_4_cap3_kit1_R1....   \n",
       "25    fastq  HI.4779.007.D707---D507.JP_101_5_cap3_kit1_R1....   \n",
       "26    fastq  HI.4779.007.D708---D507.JP_101_6_cap3_kit1_R1....   \n",
       "27    fastq  HI.4779.007.D709---D507.JP_101_7_cap4_kit1_R1....   \n",
       "28    fastq  HI.4779.007.D710---D507.JP_101_8_cap4_kit1_R1....   \n",
       "29    fastq  HI.4779.007.D711---D507.JP_101_9_cap4_kit1_R1....   \n",
       "30    fastq  HI.4779.007.D712---D507.JP_101_10_cap4_kit1_R1...   \n",
       "31    fastq  HI.4779.007.D701---D508.JP_101_11_cap4_kit1_R1...   \n",
       "32    fastq  HI.4779.007.D702---D508.JP_101_12_cap4_kit1_R1...   \n",
       "33    fastq  HI.4779.007.D703---D508.JP_101_13_cap4_kit1_R1...   \n",
       "34    fastq  HI.4779.007.D707---D508.JP_101_14_cap5_kit1_R1...   \n",
       "35    fastq  HI.4779.007.D704---D508.JP_101_15_cap5_kit1_R1...   \n",
       "36    fastq  HI.4779.007.D705---D508.JP_101_16_cap5_kit1_R1...   \n",
       "37    fastq  HI.4779.007.D706---D508.JP_101_17_cap5_kit1_R1...   \n",
       "38    fastq  HI.4779.007.D708---D508.JP_101_18_cap5_kit1_R1...   \n",
       "39    fastq  HI.4779.007.D709---D508.JP_101_19_cap5_kit1_R1...   \n",
       "40    fastq  HI.4779.007.D710---D508.JP_101_20_cap5_kit1_R1...   \n",
       "41    fastq  HI.4779.007.D711---D508.JP_p101_cap1_kit1_R1.f...   \n",
       "42    fastq  HI.4992.002.D709---D502.JP_RFmg7_cap44_kit6_R1...   \n",
       "\n",
       "                                            filename2  filename3  filename4  \\\n",
       "0   HI.4779.008.D705---D505.DF_52_20_cap3_kit1_R2....        NaN        NaN   \n",
       "1   HI.4779.008.D706---D505.DF_52_21_cap3_kit1_R2....        NaN        NaN   \n",
       "2   HI.4779.008.D707---D505.DF_52_22_cap3_kit1_R2....        NaN        NaN   \n",
       "3   HI.4779.008.D708---D505.DF_52_23_cap3_kit1_R2....        NaN        NaN   \n",
       "4   HI.4779.008.D709---D505.DF_52_24_cap3_kit1_R2....        NaN        NaN   \n",
       "5   HI.4779.008.D710---D505.DF_52_25_cap3_kit1_R2....        NaN        NaN   \n",
       "6   HI.4779.008.D711---D505.DF_52_26_cap4_kit1_R2....        NaN        NaN   \n",
       "7   HI.4779.008.D712---D505.DF_52_27_cap4_kit1_R2....        NaN        NaN   \n",
       "8   HI.4779.008.D701---D506.DF_52_28_cap4_kit1_R2....        NaN        NaN   \n",
       "9   HI.4779.008.D702---D506.DF_52_29_cap4_kit1_R2....        NaN        NaN   \n",
       "10  HI.4779.008.D703---D506.DF_52_30_cap4_kit1_R2....        NaN        NaN   \n",
       "11  HI.4779.008.D704---D506.DF_52_31_cap4_kit1_R2....        NaN        NaN   \n",
       "12  HI.4779.008.D705---D506.DF_52_32_cap4_kit1_R2....        NaN        NaN   \n",
       "13  HI.4779.008.D706---D506.DF_52_33_cap5_kit1_R2....        NaN        NaN   \n",
       "14  HI.4779.008.D707---D506.DF_52_34_cap5_kit1_R2....        NaN        NaN   \n",
       "15  HI.4779.008.D708---D506.DF_52_35_cap5_kit1_R2....        NaN        NaN   \n",
       "16  HI.4779.008.D709---D506.DF_52_36_cap5_kit1_R2....        NaN        NaN   \n",
       "17  HI.4779.008.D710---D506.DF_52_37_cap5_kit1_R2....        NaN        NaN   \n",
       "18  HI.4779.008.D711---D506.DF_52_38_cap5_kit1_R2....        NaN        NaN   \n",
       "19  HI.4779.008.D712---D506.DF_52_39_cap5_kit1_R2....        NaN        NaN   \n",
       "20  HI.4779.008.D701---D507.DF_p52_cap1_kit2_R2.fa...        NaN        NaN   \n",
       "21  HI.4779.007.D703---D507.JP_101_1_cap3_kit1_R2....        NaN        NaN   \n",
       "22  HI.4779.007.D704---D507.JP_101_2_cap3_kit1_R2....        NaN        NaN   \n",
       "23  HI.4779.007.D705---D507.JP_101_3_cap3_kit1_R2....        NaN        NaN   \n",
       "24  HI.4779.007.D706---D507.JP_101_4_cap3_kit1_R2....        NaN        NaN   \n",
       "25  HI.4779.007.D707---D507.JP_101_5_cap3_kit1_R2....        NaN        NaN   \n",
       "26  HI.4779.007.D708---D507.JP_101_6_cap3_kit1_R2....        NaN        NaN   \n",
       "27  HI.4779.007.D709---D507.JP_101_7_cap4_kit1_R2....        NaN        NaN   \n",
       "28  HI.4779.007.D710---D507.JP_101_8_cap4_kit1_R2....        NaN        NaN   \n",
       "29  HI.4779.007.D711---D507.JP_101_9_cap4_kit1_R2....        NaN        NaN   \n",
       "30  HI.4779.007.D712---D507.JP_101_10_cap4_kit1_R2...        NaN        NaN   \n",
       "31  HI.4779.007.D701---D508.JP_101_11_cap4_kit1_R2...        NaN        NaN   \n",
       "32  HI.4779.007.D702---D508.JP_101_12_cap4_kit1_R2...        NaN        NaN   \n",
       "33  HI.4779.007.D703---D508.JP_101_13_cap4_kit1_R2...        NaN        NaN   \n",
       "34  HI.4779.007.D707---D508.JP_101_14_cap5_kit1_R2...        NaN        NaN   \n",
       "35  HI.4779.007.D704---D508.JP_101_15_cap5_kit1_R2...        NaN        NaN   \n",
       "36  HI.4779.007.D705---D508.JP_101_16_cap5_kit1_R2...        NaN        NaN   \n",
       "37  HI.4779.007.D706---D508.JP_101_17_cap5_kit1_R2...        NaN        NaN   \n",
       "38  HI.4779.007.D708---D508.JP_101_18_cap5_kit1_R2...        NaN        NaN   \n",
       "39  HI.4779.007.D709---D508.JP_101_19_cap5_kit1_R2...        NaN        NaN   \n",
       "40  HI.4779.007.D710---D508.JP_101_20_cap5_kit1_R2...        NaN        NaN   \n",
       "41  HI.4779.007.D711---D508.JP_p101_cap1_kit1_R2.f...        NaN        NaN   \n",
       "42  HI.4992.002.D709---D502.JP_RFmg7_cap44_kit6_R2...        NaN        NaN   \n",
       "\n",
       "    assembly  fasta_file                   md5sum_filename  \\\n",
       "0        NaN         NaN  e5d15b40d1854dbe22f80fa424a950a3   \n",
       "1        NaN         NaN  dde217cf48666870be64700eb9c0c609   \n",
       "2        NaN         NaN  ff3c134c6e9b973f52435f9f6bf40b46   \n",
       "3        NaN         NaN  5405160b462c2bb8cc1402b001752030   \n",
       "4        NaN         NaN  a26801d87bed5cadd1e144655f557fa3   \n",
       "5        NaN         NaN  fbfe8a79f4cd7d43e3241daf0631b436   \n",
       "6        NaN         NaN  56dc37f3687405ca9c4384fe6d52d5d7   \n",
       "7        NaN         NaN  4439283a754d92079df65a8e1009286c   \n",
       "8        NaN         NaN  0ca31d096ac7b5b276fc34c45fa04805   \n",
       "9        NaN         NaN  3f2c9090a11bcc406ae47e95c79d10d6   \n",
       "10       NaN         NaN  df0c42861313609155ebe1c1e764fefc   \n",
       "11       NaN         NaN  cf46930abd6d2ad1f726baa37b42e87a   \n",
       "12       NaN         NaN  7aa3864d5e0f272bdb7c423654f6a456   \n",
       "13       NaN         NaN  3d04630962d63893dbe787553b9c3f51   \n",
       "14       NaN         NaN  70a5de1f4b4073dff74fff7258f0ddbf   \n",
       "15       NaN         NaN  a17e83687aee0e018243eee2cc1a6ba4   \n",
       "16       NaN         NaN  b2ab6085a3e5975c07e9ca8667f67879   \n",
       "17       NaN         NaN  0812d4fa8b503c47999b55ae670a8f27   \n",
       "18       NaN         NaN  56504de256a915bb62b2a0cc43269ec4   \n",
       "19       NaN         NaN  762891a748a1943baeaca449905652d2   \n",
       "20       NaN         NaN  93a7e9b5697bf934e77831c1a0485002   \n",
       "21       NaN         NaN  ad082957f1fc4b95c69d2b7bdf1ee31c   \n",
       "22       NaN         NaN  54fa6e041de38b722d55d90e231cec18   \n",
       "23       NaN         NaN  db555bc3e3045033328a7a3090127434   \n",
       "24       NaN         NaN  6e3445a2cc22d242b3303bc85388aeeb   \n",
       "25       NaN         NaN  df6e4c0ed86662e485c53b250dfb1e11   \n",
       "26       NaN         NaN  2f40fdb00ec35e0382eab86f8196f16e   \n",
       "27       NaN         NaN  67ab9a3edb538103b7a604b1bf6ae1b9   \n",
       "28       NaN         NaN  80df2eeefa4e4366f0c95710c01aa54c   \n",
       "29       NaN         NaN  130d7fd0dc1414ff2b21508ada630d4b   \n",
       "30       NaN         NaN  75cc8f8d6840c0a3c1fe115dbab8f6aa   \n",
       "31       NaN         NaN  aabc3f5d1d747f2dbb3108c26b893791   \n",
       "32       NaN         NaN  67caae3fea225c7a807097c161579d05   \n",
       "33       NaN         NaN  f270e4762e964ee2647e9d6f23390e3b   \n",
       "34       NaN         NaN  d72ea8a81ba04e82de6c07dd16f89a69   \n",
       "35       NaN         NaN  c6a7803776c8d9f23c7e61756a008ff2   \n",
       "36       NaN         NaN  9d63dfca7cd51cf6139b933469d3c7d0   \n",
       "37       NaN         NaN  85cf94e964429e966847aeef718dc41e   \n",
       "38       NaN         NaN  9c3c040fcc745012119a7e1d4660e167   \n",
       "39       NaN         NaN  309f0ad70d7dd49786dd3cf6b5375cc7   \n",
       "40       NaN         NaN  360b8921683861ad16f9c6a9e68f3f89   \n",
       "41       NaN         NaN  c78ef4f756daef61c9b2ff1823715dd9   \n",
       "42       NaN         NaN  cb94de1a82491caf1e62fe4792cb7cab   \n",
       "\n",
       "                    md5sum_filename2  \n",
       "0   ee60808baf5be0efead6d98ff4df8b57  \n",
       "1   5c5fcfc1aabfd3bd585b82accfacc7cb  \n",
       "2   cb006b5bcf0bfbf2fc2db7d47884d00b  \n",
       "3   0530c698b878fa5072be186ab376c299  \n",
       "4   ccf722a07c85e4f7dd23e78ad887b5a1  \n",
       "5   5cef6d35bf4f78fc7f2cf93dce81b176  \n",
       "6   07d132123750d7f9a10b4a48efc8da2e  \n",
       "7   d7e67ea063d959210e95f14d93b8fc06  \n",
       "8   1cab668b96a996ffd7c09f2d2dfe02d4  \n",
       "9   15ed66bb31debe7ad1487bf225b8e664  \n",
       "10  19340865fc53e595e63d7b02b723c1c1  \n",
       "11  220563fd76c44abc3df3301f7a38b916  \n",
       "12  aa26977b5946e17141ba81e84c973c94  \n",
       "13  039409788a4ec68c71fe60c575b049ac  \n",
       "14  c488d1236bf60fb09f5f58241b0d1811  \n",
       "15  7214ae872d9ebf3877a5034939e401f7  \n",
       "16  b7973b6ab309998bd78f73eef91c2ad6  \n",
       "17  600ea0e3e27810c661f4faa4bc4ce42e  \n",
       "18  5d42b8f35203e4707d24372c55c9456b  \n",
       "19  d1880109e1dfaace8f3f8049c32d771d  \n",
       "20  f753734200ce808bc1a7e5df192813a1  \n",
       "21  bf593c677390d9007294846c84cab753  \n",
       "22  2783adfa0851a4ef68a12f0f008e1e95  \n",
       "23  68667446f1306001012d80814730f16c  \n",
       "24  8eac74240e5465ac241e0b952f5f9001  \n",
       "25  56ae69379a44e7363e5cf15577436868  \n",
       "26  0bff2daf1fedb47dd33946146d725423  \n",
       "27  7b550ab972d4eb3323398b7397a30889  \n",
       "28  bd1fc32bf3d2f4d0d45863ad1b7eda97  \n",
       "29  27ceada02aa0446d4dc6aed4c64fa4b6  \n",
       "30  f68e83b5443fe2ebf29dbab620627f7d  \n",
       "31  933ae114dfc9a9761243bbbdd6378822  \n",
       "32  fd2b666f6ee543b5305725034d8c7257  \n",
       "33  ab37fd98d1578ae643a9c14f56a14ffb  \n",
       "34  1ae31f0f14de3e90ece90991ef49ebfb  \n",
       "35  a0400ecb78cf8374626726500a5525a6  \n",
       "36  da8aa4ae001507eb3e23a7d34410b4b6  \n",
       "37  4f44a1b88eed7c0eb0c5cfb30fba98c1  \n",
       "38  0c0e24d8bbb4793f976fb43e40d3ba86  \n",
       "39  b1cd39cae55e2e3fa5f1380bcc717f07  \n",
       "40  5517ac0f913201b94950a8f19fea395d  \n",
       "41  b6d7c918716125f052e67fda80575b78  \n",
       "42  342e45d7c4eb19fb61256997209c2cdb  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# fill in SRA metadata\n",
    "sra_doc = pd.concat(\n",
    "    [\n",
    "        add_to_table(datatable),\n",
    "        add_to_table(megadt)\n",
    "    ]\n",
    ").reset_index(drop=True)\n",
    "\n",
    "sra_doc['biosample_accession'] = sra_doc['library_ID'].map(biodict)\n",
    "sra_doc['md5sum_filename'] = sra_doc['filename'].map(md5s)\n",
    "sra_doc['md5sum_filename2'] = sra_doc['filename2'].map(md5s)\n",
    "assert sra_doc['md5sum_filename'].isnull().sum() == 0\n",
    "assert sra_doc['md5sum_filename2'].isnull().sum() == 0\n",
    "\n",
    "sra_doc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make sure I have unique data\n",
    "for col in ['library_ID', 'filename', 'filename2']:\n",
    "    assert luni(sra_doc[col]) == nrow(sra_doc)\n",
    "\n",
    "assert luni(sra_doc['filename'].tolist() + sra_doc['filename2'].tolist()) == nrow(sra_doc) * 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create metadata file for upload\n",
    "srafile = op.join(sra_dir, 'testdata_sra_doc.txt')\n",
    "sra_doc.to_csv(srafile, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/data/projects/pool_seq/sra_docs/testdata_submission/testdata_sra_doc.txt'"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "srafile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "dryad.append(srafile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id='biosample'></a>\n",
    "# prepare biosample metadata\n",
    "\n",
    "use this to register biosamples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/data/projects/pool_seq/sra_docs'"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sra_main"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>sample_title</th>\n",
       "      <th>bioproject_accession</th>\n",
       "      <th>organism</th>\n",
       "      <th>isolate</th>\n",
       "      <th>cultivar</th>\n",
       "      <th>ecotype</th>\n",
       "      <th>age</th>\n",
       "      <th>dev_stage</th>\n",
       "      <th>geo_loc_name</th>\n",
       "      <th>tissue</th>\n",
       "      <th>biomaterial_provider</th>\n",
       "      <th>cell_line</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>collected_by</th>\n",
       "      <th>collection_date</th>\n",
       "      <th>culture_collection</th>\n",
       "      <th>disease</th>\n",
       "      <th>disease_stage</th>\n",
       "      <th>genotype</th>\n",
       "      <th>growth_protocol</th>\n",
       "      <th>height_or_length</th>\n",
       "      <th>isolation_source</th>\n",
       "      <th>lat_lon</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>population</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>sex</th>\n",
       "      <th>specimen_voucher</th>\n",
       "      <th>temp</th>\n",
       "      <th>treatment</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [sample_name, sample_title, bioproject_accession, organism, isolate, cultivar, ecotype, age, dev_stage, geo_loc_name, tissue, biomaterial_provider, cell_line, cell_type, collected_by, collection_date, culture_collection, disease, disease_stage, genotype, growth_protocol, height_or_length, isolation_source, lat_lon, phenotype, population, sample_type, sex, specimen_voucher, temp, treatment, description]\n",
       "Index: []"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def create_bio():\n",
    "    \"\"\"Create biosample metadata using NCBI template.\"\"\"\n",
    "    bio_template = op.join(sra_main, 'Plant.1.0.txt')\n",
    "    biodf = pd.read_table(bio_template).loc[range(0)]\n",
    "    biodf.columns = [col.replace(\"*\", \"\") for col in biodf.columns]\n",
    "    return biodf\n",
    "\n",
    "create_bio()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>sample_title</th>\n",
       "      <th>bioproject_accession</th>\n",
       "      <th>organism</th>\n",
       "      <th>isolate</th>\n",
       "      <th>cultivar</th>\n",
       "      <th>ecotype</th>\n",
       "      <th>age</th>\n",
       "      <th>dev_stage</th>\n",
       "      <th>geo_loc_name</th>\n",
       "      <th>tissue</th>\n",
       "      <th>biomaterial_provider</th>\n",
       "      <th>cell_line</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>collected_by</th>\n",
       "      <th>collection_date</th>\n",
       "      <th>culture_collection</th>\n",
       "      <th>disease</th>\n",
       "      <th>disease_stage</th>\n",
       "      <th>genotype</th>\n",
       "      <th>growth_protocol</th>\n",
       "      <th>height_or_length</th>\n",
       "      <th>isolation_source</th>\n",
       "      <th>lat_lon</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>population</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>sex</th>\n",
       "      <th>specimen_voucher</th>\n",
       "      <th>temp</th>\n",
       "      <th>treatment</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DF_52_20</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DF_52_21</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DF_52_22</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_22</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DF_52_23</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DF_52_24</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>DF_52_25</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>DF_52_26</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>DF_52_27</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>DF_52_28</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_28</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>DF_52_29</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>DF_52_30</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>DF_52_31</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_31</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>DF_52_32</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>DF_52_33</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>DF_52_34</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_34</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>DF_52_35</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_35</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>DF_52_36</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_36</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>DF_52_37</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_37</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>DF_52_38</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>DF_52_39</td>\n",
       "      <td>individual diploid exome-capture of Douglas-fi...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_52_39</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>DF_p52</td>\n",
       "      <td>pooled diploid exome-capture of Douglas-fir (P...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pseudotsuga menziesii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>var. menziessii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>USA:California</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.380 N 120.670 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DF_p52</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>JP_101_1</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>JP_101_2</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>JP_101_3</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>JP_101_4</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>JP_101_5</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>JP_101_6</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>JP_101_7</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>JP_101_8</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>JP_101_9</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>JP_101_10</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>JP_101_11</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>JP_101_12</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>JP_101_13</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>JP_101_14</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>JP_101_15</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>JP_101_16</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>JP_101_17</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>JP_101_18</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>JP_101_19</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>JP_101_20</td>\n",
       "      <td>individual diploid exome-capture of jack pine ...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_101_20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>JP_p101</td>\n",
       "      <td>pooled diploid exome-capture of jack pine (Pin...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>diploid needle tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_p101</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>JP_RFmg7</td>\n",
       "      <td>haploid exome-capture of jack pine (Pinus bank...</td>\n",
       "      <td>PRJNA744263</td>\n",
       "      <td>Pinus banksiana</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sakskatchewan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>juvenile</td>\n",
       "      <td>Canada:Saskatchewan</td>\n",
       "      <td>haploid megagametophyte tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.083 N 107.250 W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JP_RFmg7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample_name                                       sample_title  \\\n",
       "0     DF_52_20  individual diploid exome-capture of Douglas-fi...   \n",
       "1     DF_52_21  individual diploid exome-capture of Douglas-fi...   \n",
       "2     DF_52_22  individual diploid exome-capture of Douglas-fi...   \n",
       "3     DF_52_23  individual diploid exome-capture of Douglas-fi...   \n",
       "4     DF_52_24  individual diploid exome-capture of Douglas-fi...   \n",
       "5     DF_52_25  individual diploid exome-capture of Douglas-fi...   \n",
       "6     DF_52_26  individual diploid exome-capture of Douglas-fi...   \n",
       "7     DF_52_27  individual diploid exome-capture of Douglas-fi...   \n",
       "8     DF_52_28  individual diploid exome-capture of Douglas-fi...   \n",
       "9     DF_52_29  individual diploid exome-capture of Douglas-fi...   \n",
       "10    DF_52_30  individual diploid exome-capture of Douglas-fi...   \n",
       "11    DF_52_31  individual diploid exome-capture of Douglas-fi...   \n",
       "12    DF_52_32  individual diploid exome-capture of Douglas-fi...   \n",
       "13    DF_52_33  individual diploid exome-capture of Douglas-fi...   \n",
       "14    DF_52_34  individual diploid exome-capture of Douglas-fi...   \n",
       "15    DF_52_35  individual diploid exome-capture of Douglas-fi...   \n",
       "16    DF_52_36  individual diploid exome-capture of Douglas-fi...   \n",
       "17    DF_52_37  individual diploid exome-capture of Douglas-fi...   \n",
       "18    DF_52_38  individual diploid exome-capture of Douglas-fi...   \n",
       "19    DF_52_39  individual diploid exome-capture of Douglas-fi...   \n",
       "20      DF_p52  pooled diploid exome-capture of Douglas-fir (P...   \n",
       "21    JP_101_1  individual diploid exome-capture of jack pine ...   \n",
       "22    JP_101_2  individual diploid exome-capture of jack pine ...   \n",
       "23    JP_101_3  individual diploid exome-capture of jack pine ...   \n",
       "24    JP_101_4  individual diploid exome-capture of jack pine ...   \n",
       "25    JP_101_5  individual diploid exome-capture of jack pine ...   \n",
       "26    JP_101_6  individual diploid exome-capture of jack pine ...   \n",
       "27    JP_101_7  individual diploid exome-capture of jack pine ...   \n",
       "28    JP_101_8  individual diploid exome-capture of jack pine ...   \n",
       "29    JP_101_9  individual diploid exome-capture of jack pine ...   \n",
       "30   JP_101_10  individual diploid exome-capture of jack pine ...   \n",
       "31   JP_101_11  individual diploid exome-capture of jack pine ...   \n",
       "32   JP_101_12  individual diploid exome-capture of jack pine ...   \n",
       "33   JP_101_13  individual diploid exome-capture of jack pine ...   \n",
       "34   JP_101_14  individual diploid exome-capture of jack pine ...   \n",
       "35   JP_101_15  individual diploid exome-capture of jack pine ...   \n",
       "36   JP_101_16  individual diploid exome-capture of jack pine ...   \n",
       "37   JP_101_17  individual diploid exome-capture of jack pine ...   \n",
       "38   JP_101_18  individual diploid exome-capture of jack pine ...   \n",
       "39   JP_101_19  individual diploid exome-capture of jack pine ...   \n",
       "40   JP_101_20  individual diploid exome-capture of jack pine ...   \n",
       "41     JP_p101  pooled diploid exome-capture of jack pine (Pin...   \n",
       "42    JP_RFmg7  haploid exome-capture of jack pine (Pinus bank...   \n",
       "\n",
       "   bioproject_accession               organism isolate cultivar  \\\n",
       "0           PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "1           PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "2           PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "3           PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "4           PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "5           PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "6           PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "7           PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "8           PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "9           PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "10          PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "11          PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "12          PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "13          PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "14          PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "15          PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "16          PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "17          PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "18          PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "19          PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "20          PRJNA744263  Pseudotsuga menziesii     NaN      NaN   \n",
       "21          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "22          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "23          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "24          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "25          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "26          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "27          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "28          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "29          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "30          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "31          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "32          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "33          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "34          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "35          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "36          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "37          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "38          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "39          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "40          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "41          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "42          PRJNA744263        Pinus banksiana     NaN      NaN   \n",
       "\n",
       "            ecotype  age dev_stage         geo_loc_name  \\\n",
       "0   var. menziessii  NaN  juvenile       USA:California   \n",
       "1   var. menziessii  NaN  juvenile       USA:California   \n",
       "2   var. menziessii  NaN  juvenile       USA:California   \n",
       "3   var. menziessii  NaN  juvenile       USA:California   \n",
       "4   var. menziessii  NaN  juvenile       USA:California   \n",
       "5   var. menziessii  NaN  juvenile       USA:California   \n",
       "6   var. menziessii  NaN  juvenile       USA:California   \n",
       "7   var. menziessii  NaN  juvenile       USA:California   \n",
       "8   var. menziessii  NaN  juvenile       USA:California   \n",
       "9   var. menziessii  NaN  juvenile       USA:California   \n",
       "10  var. menziessii  NaN  juvenile       USA:California   \n",
       "11  var. menziessii  NaN  juvenile       USA:California   \n",
       "12  var. menziessii  NaN  juvenile       USA:California   \n",
       "13  var. menziessii  NaN  juvenile       USA:California   \n",
       "14  var. menziessii  NaN  juvenile       USA:California   \n",
       "15  var. menziessii  NaN  juvenile       USA:California   \n",
       "16  var. menziessii  NaN  juvenile       USA:California   \n",
       "17  var. menziessii  NaN  juvenile       USA:California   \n",
       "18  var. menziessii  NaN  juvenile       USA:California   \n",
       "19  var. menziessii  NaN  juvenile       USA:California   \n",
       "20  var. menziessii  NaN  juvenile       USA:California   \n",
       "21    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "22    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "23    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "24    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "25    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "26    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "27    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "28    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "29    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "30    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "31    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "32    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "33    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "34    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "35    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "36    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "37    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "38    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "39    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "40    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "41    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "42    Sakskatchewan  NaN  juvenile  Canada:Saskatchewan   \n",
       "\n",
       "                            tissue biomaterial_provider cell_line cell_type  \\\n",
       "0            diploid needle tissue                  NaN       NaN       NaN   \n",
       "1            diploid needle tissue                  NaN       NaN       NaN   \n",
       "2            diploid needle tissue                  NaN       NaN       NaN   \n",
       "3            diploid needle tissue                  NaN       NaN       NaN   \n",
       "4            diploid needle tissue                  NaN       NaN       NaN   \n",
       "5            diploid needle tissue                  NaN       NaN       NaN   \n",
       "6            diploid needle tissue                  NaN       NaN       NaN   \n",
       "7            diploid needle tissue                  NaN       NaN       NaN   \n",
       "8            diploid needle tissue                  NaN       NaN       NaN   \n",
       "9            diploid needle tissue                  NaN       NaN       NaN   \n",
       "10           diploid needle tissue                  NaN       NaN       NaN   \n",
       "11           diploid needle tissue                  NaN       NaN       NaN   \n",
       "12           diploid needle tissue                  NaN       NaN       NaN   \n",
       "13           diploid needle tissue                  NaN       NaN       NaN   \n",
       "14           diploid needle tissue                  NaN       NaN       NaN   \n",
       "15           diploid needle tissue                  NaN       NaN       NaN   \n",
       "16           diploid needle tissue                  NaN       NaN       NaN   \n",
       "17           diploid needle tissue                  NaN       NaN       NaN   \n",
       "18           diploid needle tissue                  NaN       NaN       NaN   \n",
       "19           diploid needle tissue                  NaN       NaN       NaN   \n",
       "20           diploid needle tissue                  NaN       NaN       NaN   \n",
       "21           diploid needle tissue                  NaN       NaN       NaN   \n",
       "22           diploid needle tissue                  NaN       NaN       NaN   \n",
       "23           diploid needle tissue                  NaN       NaN       NaN   \n",
       "24           diploid needle tissue                  NaN       NaN       NaN   \n",
       "25           diploid needle tissue                  NaN       NaN       NaN   \n",
       "26           diploid needle tissue                  NaN       NaN       NaN   \n",
       "27           diploid needle tissue                  NaN       NaN       NaN   \n",
       "28           diploid needle tissue                  NaN       NaN       NaN   \n",
       "29           diploid needle tissue                  NaN       NaN       NaN   \n",
       "30           diploid needle tissue                  NaN       NaN       NaN   \n",
       "31           diploid needle tissue                  NaN       NaN       NaN   \n",
       "32           diploid needle tissue                  NaN       NaN       NaN   \n",
       "33           diploid needle tissue                  NaN       NaN       NaN   \n",
       "34           diploid needle tissue                  NaN       NaN       NaN   \n",
       "35           diploid needle tissue                  NaN       NaN       NaN   \n",
       "36           diploid needle tissue                  NaN       NaN       NaN   \n",
       "37           diploid needle tissue                  NaN       NaN       NaN   \n",
       "38           diploid needle tissue                  NaN       NaN       NaN   \n",
       "39           diploid needle tissue                  NaN       NaN       NaN   \n",
       "40           diploid needle tissue                  NaN       NaN       NaN   \n",
       "41           diploid needle tissue                  NaN       NaN       NaN   \n",
       "42  haploid megagametophyte tissue                  NaN       NaN       NaN   \n",
       "\n",
       "   collected_by collection_date culture_collection disease disease_stage  \\\n",
       "0           NaN             NaN                NaN     NaN           NaN   \n",
       "1           NaN             NaN                NaN     NaN           NaN   \n",
       "2           NaN             NaN                NaN     NaN           NaN   \n",
       "3           NaN             NaN                NaN     NaN           NaN   \n",
       "4           NaN             NaN                NaN     NaN           NaN   \n",
       "5           NaN             NaN                NaN     NaN           NaN   \n",
       "6           NaN             NaN                NaN     NaN           NaN   \n",
       "7           NaN             NaN                NaN     NaN           NaN   \n",
       "8           NaN             NaN                NaN     NaN           NaN   \n",
       "9           NaN             NaN                NaN     NaN           NaN   \n",
       "10          NaN             NaN                NaN     NaN           NaN   \n",
       "11          NaN             NaN                NaN     NaN           NaN   \n",
       "12          NaN             NaN                NaN     NaN           NaN   \n",
       "13          NaN             NaN                NaN     NaN           NaN   \n",
       "14          NaN             NaN                NaN     NaN           NaN   \n",
       "15          NaN             NaN                NaN     NaN           NaN   \n",
       "16          NaN             NaN                NaN     NaN           NaN   \n",
       "17          NaN             NaN                NaN     NaN           NaN   \n",
       "18          NaN             NaN                NaN     NaN           NaN   \n",
       "19          NaN             NaN                NaN     NaN           NaN   \n",
       "20          NaN             NaN                NaN     NaN           NaN   \n",
       "21          NaN             NaN                NaN     NaN           NaN   \n",
       "22          NaN             NaN                NaN     NaN           NaN   \n",
       "23          NaN             NaN                NaN     NaN           NaN   \n",
       "24          NaN             NaN                NaN     NaN           NaN   \n",
       "25          NaN             NaN                NaN     NaN           NaN   \n",
       "26          NaN             NaN                NaN     NaN           NaN   \n",
       "27          NaN             NaN                NaN     NaN           NaN   \n",
       "28          NaN             NaN                NaN     NaN           NaN   \n",
       "29          NaN             NaN                NaN     NaN           NaN   \n",
       "30          NaN             NaN                NaN     NaN           NaN   \n",
       "31          NaN             NaN                NaN     NaN           NaN   \n",
       "32          NaN             NaN                NaN     NaN           NaN   \n",
       "33          NaN             NaN                NaN     NaN           NaN   \n",
       "34          NaN             NaN                NaN     NaN           NaN   \n",
       "35          NaN             NaN                NaN     NaN           NaN   \n",
       "36          NaN             NaN                NaN     NaN           NaN   \n",
       "37          NaN             NaN                NaN     NaN           NaN   \n",
       "38          NaN             NaN                NaN     NaN           NaN   \n",
       "39          NaN             NaN                NaN     NaN           NaN   \n",
       "40          NaN             NaN                NaN     NaN           NaN   \n",
       "41          NaN             NaN                NaN     NaN           NaN   \n",
       "42          NaN             NaN                NaN     NaN           NaN   \n",
       "\n",
       "   genotype growth_protocol height_or_length isolation_source  \\\n",
       "0       NaN             NaN              NaN              NaN   \n",
       "1       NaN             NaN              NaN              NaN   \n",
       "2       NaN             NaN              NaN              NaN   \n",
       "3       NaN             NaN              NaN              NaN   \n",
       "4       NaN             NaN              NaN              NaN   \n",
       "5       NaN             NaN              NaN              NaN   \n",
       "6       NaN             NaN              NaN              NaN   \n",
       "7       NaN             NaN              NaN              NaN   \n",
       "8       NaN             NaN              NaN              NaN   \n",
       "9       NaN             NaN              NaN              NaN   \n",
       "10      NaN             NaN              NaN              NaN   \n",
       "11      NaN             NaN              NaN              NaN   \n",
       "12      NaN             NaN              NaN              NaN   \n",
       "13      NaN             NaN              NaN              NaN   \n",
       "14      NaN             NaN              NaN              NaN   \n",
       "15      NaN             NaN              NaN              NaN   \n",
       "16      NaN             NaN              NaN              NaN   \n",
       "17      NaN             NaN              NaN              NaN   \n",
       "18      NaN             NaN              NaN              NaN   \n",
       "19      NaN             NaN              NaN              NaN   \n",
       "20      NaN             NaN              NaN              NaN   \n",
       "21      NaN             NaN              NaN              NaN   \n",
       "22      NaN             NaN              NaN              NaN   \n",
       "23      NaN             NaN              NaN              NaN   \n",
       "24      NaN             NaN              NaN              NaN   \n",
       "25      NaN             NaN              NaN              NaN   \n",
       "26      NaN             NaN              NaN              NaN   \n",
       "27      NaN             NaN              NaN              NaN   \n",
       "28      NaN             NaN              NaN              NaN   \n",
       "29      NaN             NaN              NaN              NaN   \n",
       "30      NaN             NaN              NaN              NaN   \n",
       "31      NaN             NaN              NaN              NaN   \n",
       "32      NaN             NaN              NaN              NaN   \n",
       "33      NaN             NaN              NaN              NaN   \n",
       "34      NaN             NaN              NaN              NaN   \n",
       "35      NaN             NaN              NaN              NaN   \n",
       "36      NaN             NaN              NaN              NaN   \n",
       "37      NaN             NaN              NaN              NaN   \n",
       "38      NaN             NaN              NaN              NaN   \n",
       "39      NaN             NaN              NaN              NaN   \n",
       "40      NaN             NaN              NaN              NaN   \n",
       "41      NaN             NaN              NaN              NaN   \n",
       "42      NaN             NaN              NaN              NaN   \n",
       "\n",
       "               lat_lon phenotype population sample_type  sex specimen_voucher  \\\n",
       "0   39.380 N 120.670 W       NaN   DF_52_20         NaN  NaN              NaN   \n",
       "1   39.380 N 120.670 W       NaN   DF_52_21         NaN  NaN              NaN   \n",
       "2   39.380 N 120.670 W       NaN   DF_52_22         NaN  NaN              NaN   \n",
       "3   39.380 N 120.670 W       NaN   DF_52_23         NaN  NaN              NaN   \n",
       "4   39.380 N 120.670 W       NaN   DF_52_24         NaN  NaN              NaN   \n",
       "5   39.380 N 120.670 W       NaN   DF_52_25         NaN  NaN              NaN   \n",
       "6   39.380 N 120.670 W       NaN   DF_52_26         NaN  NaN              NaN   \n",
       "7   39.380 N 120.670 W       NaN   DF_52_27         NaN  NaN              NaN   \n",
       "8   39.380 N 120.670 W       NaN   DF_52_28         NaN  NaN              NaN   \n",
       "9   39.380 N 120.670 W       NaN   DF_52_29         NaN  NaN              NaN   \n",
       "10  39.380 N 120.670 W       NaN   DF_52_30         NaN  NaN              NaN   \n",
       "11  39.380 N 120.670 W       NaN   DF_52_31         NaN  NaN              NaN   \n",
       "12  39.380 N 120.670 W       NaN   DF_52_32         NaN  NaN              NaN   \n",
       "13  39.380 N 120.670 W       NaN   DF_52_33         NaN  NaN              NaN   \n",
       "14  39.380 N 120.670 W       NaN   DF_52_34         NaN  NaN              NaN   \n",
       "15  39.380 N 120.670 W       NaN   DF_52_35         NaN  NaN              NaN   \n",
       "16  39.380 N 120.670 W       NaN   DF_52_36         NaN  NaN              NaN   \n",
       "17  39.380 N 120.670 W       NaN   DF_52_37         NaN  NaN              NaN   \n",
       "18  39.380 N 120.670 W       NaN   DF_52_38         NaN  NaN              NaN   \n",
       "19  39.380 N 120.670 W       NaN   DF_52_39         NaN  NaN              NaN   \n",
       "20  39.380 N 120.670 W       NaN     DF_p52         NaN  NaN              NaN   \n",
       "21  54.083 N 107.250 W       NaN   JP_101_1         NaN  NaN              NaN   \n",
       "22  54.083 N 107.250 W       NaN   JP_101_2         NaN  NaN              NaN   \n",
       "23  54.083 N 107.250 W       NaN   JP_101_3         NaN  NaN              NaN   \n",
       "24  54.083 N 107.250 W       NaN   JP_101_4         NaN  NaN              NaN   \n",
       "25  54.083 N 107.250 W       NaN   JP_101_5         NaN  NaN              NaN   \n",
       "26  54.083 N 107.250 W       NaN   JP_101_6         NaN  NaN              NaN   \n",
       "27  54.083 N 107.250 W       NaN   JP_101_7         NaN  NaN              NaN   \n",
       "28  54.083 N 107.250 W       NaN   JP_101_8         NaN  NaN              NaN   \n",
       "29  54.083 N 107.250 W       NaN   JP_101_9         NaN  NaN              NaN   \n",
       "30  54.083 N 107.250 W       NaN  JP_101_10         NaN  NaN              NaN   \n",
       "31  54.083 N 107.250 W       NaN  JP_101_11         NaN  NaN              NaN   \n",
       "32  54.083 N 107.250 W       NaN  JP_101_12         NaN  NaN              NaN   \n",
       "33  54.083 N 107.250 W       NaN  JP_101_13         NaN  NaN              NaN   \n",
       "34  54.083 N 107.250 W       NaN  JP_101_14         NaN  NaN              NaN   \n",
       "35  54.083 N 107.250 W       NaN  JP_101_15         NaN  NaN              NaN   \n",
       "36  54.083 N 107.250 W       NaN  JP_101_16         NaN  NaN              NaN   \n",
       "37  54.083 N 107.250 W       NaN  JP_101_17         NaN  NaN              NaN   \n",
       "38  54.083 N 107.250 W       NaN  JP_101_18         NaN  NaN              NaN   \n",
       "39  54.083 N 107.250 W       NaN  JP_101_19         NaN  NaN              NaN   \n",
       "40  54.083 N 107.250 W       NaN  JP_101_20         NaN  NaN              NaN   \n",
       "41  54.083 N 107.250 W       NaN    JP_p101         NaN  NaN              NaN   \n",
       "42  54.083 N 107.250 W       NaN   JP_RFmg7         NaN  NaN              NaN   \n",
       "\n",
       "   temp treatment description  \n",
       "0   NaN       NaN         NaN  \n",
       "1   NaN       NaN         NaN  \n",
       "2   NaN       NaN         NaN  \n",
       "3   NaN       NaN         NaN  \n",
       "4   NaN       NaN         NaN  \n",
       "5   NaN       NaN         NaN  \n",
       "6   NaN       NaN         NaN  \n",
       "7   NaN       NaN         NaN  \n",
       "8   NaN       NaN         NaN  \n",
       "9   NaN       NaN         NaN  \n",
       "10  NaN       NaN         NaN  \n",
       "11  NaN       NaN         NaN  \n",
       "12  NaN       NaN         NaN  \n",
       "13  NaN       NaN         NaN  \n",
       "14  NaN       NaN         NaN  \n",
       "15  NaN       NaN         NaN  \n",
       "16  NaN       NaN         NaN  \n",
       "17  NaN       NaN         NaN  \n",
       "18  NaN       NaN         NaN  \n",
       "19  NaN       NaN         NaN  \n",
       "20  NaN       NaN         NaN  \n",
       "21  NaN       NaN         NaN  \n",
       "22  NaN       NaN         NaN  \n",
       "23  NaN       NaN         NaN  \n",
       "24  NaN       NaN         NaN  \n",
       "25  NaN       NaN         NaN  \n",
       "26  NaN       NaN         NaN  \n",
       "27  NaN       NaN         NaN  \n",
       "28  NaN       NaN         NaN  \n",
       "29  NaN       NaN         NaN  \n",
       "30  NaN       NaN         NaN  \n",
       "31  NaN       NaN         NaN  \n",
       "32  NaN       NaN         NaN  \n",
       "33  NaN       NaN         NaN  \n",
       "34  NaN       NaN         NaN  \n",
       "35  NaN       NaN         NaN  \n",
       "36  NaN       NaN         NaN  \n",
       "37  NaN       NaN         NaN  \n",
       "38  NaN       NaN         NaN  \n",
       "39  NaN       NaN         NaN  \n",
       "40  NaN       NaN         NaN  \n",
       "41  NaN       NaN         NaN  \n",
       "42  NaN       NaN         NaN  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# fill in biosample metadata\n",
    "bio = create_bio()\n",
    "for row in sra_doc.index:\n",
    "    samp,title = sra_doc.loc[row, ['library_ID', 'title']]\n",
    "    \n",
    "    if 'DF' in samp:\n",
    "        spp = 'Pseudotsuga menziesii'\n",
    "        geo = 'USA:California'\n",
    "        latlon = '39.380 N 120.670 W'\n",
    "        ecotype = 'var. menziessii'\n",
    "    else:\n",
    "        spp = 'Pinus banksiana'\n",
    "        geo = 'Canada:Saskatchewan'\n",
    "        latlon = '54.083 N 107.250 W'\n",
    "        ecotype = 'Sakskatchewan'\n",
    "\n",
    "    bio.loc[row, 'sample_name'] = samp\n",
    "    bio.loc[row, 'sample_title'] = title\n",
    "    bio.loc[row, 'organism'] = spp\n",
    "    bio.loc[row, 'geo_loc_name'] = geo\n",
    "    bio.loc[row, 'tissue'] = title.split(\": \")[-1]\n",
    "    bio.loc[row, 'population'] = samp\n",
    "    bio.loc[row, 'lat_lon'] = latlon\n",
    "    bio.loc[row, 'ecotype'] = ecotype\n",
    "\n",
    "bio['dev_stage'] = 'juvenile'\n",
    "bio['bioproject_accession'] = 'PRJNA744263'\n",
    "\n",
    "bio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save\n",
    "biofile = op.join(sra_dir, 'testdata_biosample.txt')\n",
    "bio.to_csv(biofile, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "dryad.append(biofile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/data/projects/pool_seq/sra_docs/testdata_submission/testdata_biosample.txt'"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biofile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/data/projects/pool_seq/non-pangenome/gatk_diploid_testdata/JP_i101-gatk/datatable.txt',\n",
       " '/data/projects/pool_seq/non-pangenome/varscan_mega/JP_RFmg7/datatable.txt',\n",
       " '/lu213/brandon.lind/data/testdir/dryad_testdata/DF_i52_filtered_concatenated_snps_max-missing_table_biallelic-only_p000_translated.txt.gz',\n",
       " '/lu213/brandon.lind/data/testdir/dryad_testdata/DF_mega-varscan_all_bedfiles_SNP_paralog_snps.txt.gz',\n",
       " '/lu213/brandon.lind/data/testdir/dryad_testdata/DF_p52-varscan_all_bedfiles_SNP.txt.gz',\n",
       " '/lu213/brandon.lind/data/testdir/dryad_testdata/JP_RFmg7-varscan_all_bedfiles_SNP_paralog_snps_translated.txt.gz',\n",
       " '/lu213/brandon.lind/data/testdir/dryad_testdata/JP_i101_filtered_concatenated_snps_max-missing_table_biallelic-only_translated.txt.gz',\n",
       " '/lu213/brandon.lind/data/testdir/dryad_testdata/JP_pooled-varscan_all_bedfiles_SNP_translated.txt.gz',\n",
       " '/data/projects/pool_seq/sra_docs/testdata_submission/testdata_sra_doc.txt',\n",
       " '/data/projects/pool_seq/sra_docs/testdata_submission/testdata_biosample.txt']"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dryad"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(10, 10)"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(dryad), luni(dryad)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id='transfer'></a>\n",
    "# transfer files to Short Read Archive via ftp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "def connect(password_file):\n",
    "    \"\"\"Connect to NCBI SRA ftp.\"\"\"\n",
    "    import ftplib\n",
    "    \n",
    "    user,password = read(password_file, lines=True)\n",
    "    ftp = ftplib.FTP('ftp-private.ncbi.nlm.nih.gov', user, password)\n",
    "    ftp.set_pasv(True)\n",
    "    \n",
    "    return ftp\n",
    "\n",
    "def upload_file(f, password_file):\n",
    "    \"\"\"Upload file, `f`, to ftp.\"\"\"\n",
    "    from os import path as op\n",
    "    import os\n",
    "    \n",
    "    ftp = connect(password_file)\n",
    "    ftp.cwd('uploads/lind.brandon.m_gmail.com_oR317JsQ/coadaptree_testdata_files')\n",
    "\n",
    "    with open(f, 'rb') as file:\n",
    "        ftp.storbinary(f\"STOR {op.basename(f)}\", file)\n",
    "\n",
    "    ftp.quit()\n",
    "    \n",
    "    return op.basename(f)\n",
    "\n",
    "dview['connect'] = connect\n",
    "dview['read'] = read"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1m\n",
      "Watching 172 jobs ...\u001b[0m\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 172/172 [21:20<00:00,  7.44s/it] \n"
     ]
    }
   ],
   "source": [
    "# upload individual and pool-seq fastqs\n",
    "user_passfile = '/data/projects/pool_seq/non-pangenome/gatk_diploid_testdata/testdata_sra_password'\n",
    "\n",
    "jobs = []\n",
    "for f in sra_transfer_files:\n",
    "    jobs.append(\n",
    "        lview.apply_async(\n",
    "            upload_file, *(f, user_passfile)\n",
    "        )\n",
    "    )\n",
    "    \n",
    "watch_async(jobs)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# assert successful SRA transfer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "def check_transfers():\n",
    "    ftp = connect(user_passfile)\n",
    "    ftp.cwd('uploads/lind.brandon.m_gmail.com_oR317JsQ/coadaptree_testdata_files')\n",
    "    files = ftp.nlst()\n",
    "    return files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'HI.4779.008.D705---D505.DF_52_20_cap3_kit1_R1.fastq.gz'"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dsts = [j.r for j in jobs]\n",
    "dsts[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "172"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files = check_transfers()\n",
    "\n",
    "len(files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "# assert all transfer files present in the ftp directory?\n",
    "assert all([op.basename(f) in files for f in sra_transfer_files])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make sure all the files in the metadata are in the ftp directory\n",
    "file_checks = sra_doc['filename'].tolist() + sra_doc['filename2'].tolist()\n",
    "assert all([f in files for f in file_checks])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "86"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "luni(file_checks)  # 172 / 2 = 86 (md5s + fastqs; md5s will be deleted)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "[os.remove(user_passfile)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rsync -azv yeaman03:/data/projects/pool_seq/non-pangenome/gatk_diploid_testdata/JP_i101-gatk/datatable.txt datatable.txt --progress \n",
      "\n",
      "rsync -azv yeaman03:/data/projects/pool_seq/non-pangenome/varscan_mega/JP_RFmg7/datatable.txt datatable.txt --progress \n",
      "\n",
      "rsync -azv yeaman03:/lu213/brandon.lind/data/testdir/dryad_testdata/DF_i52_filtered_concatenated_snps_max-missing_table_biallelic-only_p000_translated.txt.gz DF_i52_filtered_concatenated_snps_max-missing_table_biallelic-only_p000_translated.txt.gz --progress \n",
      "\n",
      "rsync -azv yeaman03:/lu213/brandon.lind/data/testdir/dryad_testdata/DF_mega-varscan_all_bedfiles_SNP_paralog_snps.txt.gz DF_mega-varscan_all_bedfiles_SNP_paralog_snps.txt.gz --progress \n",
      "\n",
      "rsync -azv yeaman03:/lu213/brandon.lind/data/testdir/dryad_testdata/DF_p52-varscan_all_bedfiles_SNP.txt.gz DF_p52-varscan_all_bedfiles_SNP.txt.gz --progress \n",
      "\n",
      "rsync -azv yeaman03:/lu213/brandon.lind/data/testdir/dryad_testdata/JP_RFmg7-varscan_all_bedfiles_SNP_paralog_snps_translated.txt.gz JP_RFmg7-varscan_all_bedfiles_SNP_paralog_snps_translated.txt.gz --progress \n",
      "\n",
      "rsync -azv yeaman03:/lu213/brandon.lind/data/testdir/dryad_testdata/JP_i101_filtered_concatenated_snps_max-missing_table_biallelic-only_translated.txt.gz JP_i101_filtered_concatenated_snps_max-missing_table_biallelic-only_translated.txt.gz --progress \n",
      "\n",
      "rsync -azv yeaman03:/lu213/brandon.lind/data/testdir/dryad_testdata/JP_pooled-varscan_all_bedfiles_SNP_translated.txt.gz JP_pooled-varscan_all_bedfiles_SNP_translated.txt.gz --progress \n",
      "\n",
      "rsync -azv yeaman03:/data/projects/pool_seq/sra_docs/testdata_submission/testdata_sra_doc.txt testdata_sra_doc.txt --progress \n",
      "\n",
      "rsync -azv yeaman03:/data/projects/pool_seq/sra_docs/testdata_submission/testdata_biosample.txt testdata_biosample.txt --progress \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# download dryad files to upload to dryad\n",
    "for f in dryad:\n",
    "    print('rsync -azv', f'yeaman03:{f}', op.basename(f), '--progress', '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Brandon Lind July 7, 2021\n",
      "\n",
      "All .txt and .txt.gz files are tab-delimited (once decompressed).\n",
      "\n",
      "haploid_pipeline_datatable.txt - this is the `datatable.txt` file needed to configure the run of our jack pine haploid megagametophyte data through the Varscan Pipeline\n",
      "\n",
      "pooled_individual_pipeline_datatable.txt - this is the `datatable.txt` file needed to configure the runs of our jack pine and Douglas-fir pooled and individual sequence data through both the Varscan Pipeline and the GATK pipeline\n",
      "\n",
      "DF_i52_filtered_concatenated_snps_max-missing_table_biallelic-only_p000_translated.txt.gz - the SNP calls from our GATK Pipeline for Douglas-fir individual sequence data\n",
      "\n",
      "DF_mega-varscan_all_bedfiles_SNP_paralog_snps.txt.gz - the heterozygous SNP calls for sites output from the Varscan Pipeline for our haploid Douglas-fir data\n",
      "\n",
      "DF_p52-varscan_all_bedfiles_SNP.txt.gz - the SNP calls for our pooled Douglas-fir data from the Varscan Pipeline\n",
      "\n",
      "JP_RFmg7-varscan_all_bedfiles_SNP_paralog_snps_translated.txt.gz - the heterozygous SNP calls for sites output from the Varscan Pipeline for our haploid jack pine data\n",
      "\n",
      "JP_i101_filtered_concatenated_snps_max-missing_table_biallelic-only_translated.txt.gz - the SNP calls from our GATK Pipeline for jack pine individual sequence data\n",
      "\n",
      "JP_pooled-varscan_all_bedfiles_SNP_translated.txt.gz - the SNP calls for our pooled jack pine data from the Varscan Pipeline\n",
      "\n",
      "testdata_sra_doc.txt - NCBI Short Read Archive metadata\n",
      "\n",
      "testdata_biosample.txt - NCBI Biosample metadata\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# write dryad readme\n",
    "README = op.join(sra_main, 'dryad_README.txt')\n",
    "\n",
    "with open(README, 'w') as readme:\n",
    "    readme.write('''Brandon Lind July 7, 2021\n",
    "\n",
    "All .txt and .txt.gz files are tab-delimited (once decompressed).\n",
    "\n",
    "haploid_pipeline_datatable.txt - this is the `datatable.txt` file needed to configure the run of our jack pine haploid megagametophyte data through the Varscan Pipeline\n",
    "\n",
    "pooled_individual_pipeline_datatable.txt - this is the `datatable.txt` file needed to configure the runs of our jack pine and Douglas-fir pooled and individual sequence data through both the Varscan Pipeline and the GATK pipeline\n",
    "\n",
    "DF_i52_filtered_concatenated_snps_max-missing_table_biallelic-only_p000_translated.txt.gz - the SNP calls from our GATK Pipeline for Douglas-fir individual sequence data\n",
    "\n",
    "DF_mega-varscan_all_bedfiles_SNP_paralog_snps.txt.gz - the heterozygous SNP calls for sites output from the Varscan Pipeline for our haploid Douglas-fir data\n",
    "\n",
    "DF_p52-varscan_all_bedfiles_SNP.txt.gz - the SNP calls for our pooled Douglas-fir data from the Varscan Pipeline\n",
    "\n",
    "JP_RFmg7-varscan_all_bedfiles_SNP_paralog_snps_translated.txt.gz - the heterozygous SNP calls for sites output from the Varscan Pipeline for our haploid jack pine data\n",
    "\n",
    "JP_i101_filtered_concatenated_snps_max-missing_table_biallelic-only_translated.txt.gz - the SNP calls from our GATK Pipeline for jack pine individual sequence data\n",
    "\n",
    "JP_pooled-varscan_all_bedfiles_SNP_translated.txt.gz - the SNP calls for our pooled jack pine data from the Varscan Pipeline\n",
    "\n",
    "testdata_sra_doc.txt - NCBI Short Read Archive metadata\n",
    "\n",
    "testdata_biosample.txt - NCBI Biosample metadata\n",
    "''')\n",
    "    \n",
    "print(read(README, lines=False))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/data/projects/pool_seq/sra_docs/dryad_README.txt'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "README"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
